### ISSUE TITLE
Update Axios peer dependency

### ISSUE THREAD

**Issue by christian-bromann** [zhaosiyang/axios-observable#27]
I have issues using `axios-observable` with latest Axios installed. To reproduce:

```sh
$ mkdir /tmp/deptest
$ cd /tmp/deptest
$ yarn add axios axios-observable rxjs
```

Which results in:

```
yarn add v1.22.17
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > axios-observable@1.3.0" has incorrect peer dependency "axios@^0.21.1".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
success Saved 4 new dependencies.
info Direct dependencies
├─ axios-observable@1.3.0
├─ axios@0.24.0
└─ rxjs@7.4.0
info All dependencies
├─ axios-observable@1.3.0
├─ axios@0.24.0
├─ follow-redirects@1.14.6
└─ rxjs@7.4.0
✨  Done in 0.80
```

You can see the warning:
> warning " > axios-observable@1.3.0" has incorrect peer dependency "axios@^0.21.1".

Which is odd to me because the version range is setup correctly using `^`. Not quite sure why it is still failing.

**Comment by dargmuesli** [comment]
`^0.21.1` means that any versions with a higher **patch** level are allowed (1 and up). The **minor** version is preceded by a 0 so it becomes the "major" version (21).

**Comment by whazor** [comment]
I think it is important we bump the version as the older Axios version gives security audit warnings. My PR from #28 uses `>=0.21.1` instead of `^0.21.1` allowing you to install the latest Axios yourself. For typing I had copy over an updated type from the latest Axios, I am not sure if that type still works on older versions of Axios but for sure works on the latest `0.24.0`. So maybe it is better to set the peerDependency higher.

**Comment by zhaosiyang** [comment]
Thanks for the feedback and my apologies for the late reply. 
Fixed in version 1.4.0
